Source/player.cpp

```diff
@@ -67,7 +67,7 @@ void WalkUpwards(int pnum, const DirectionSettings &walkParams)
 {
 	auto &player = plr[pnum];
 	dPlayer[player.position.future.x][player.position.future.y] = -(pnum + 1);
-	player.position.temp = walkParams.tileAdd;
+	player.position.temp = { walkParams.tileAdd.deltaX, walkParams.tileAdd.deltaY };

```

Comment:
As I said in discord. In this case `position.temp` is used as a `Displacement`.
[This](https://github.com/diasurgical/devilutionX/pull/2259/files#diff-71461699ae2f94f652e6cba2136d3c696151ba3a6c0e357eaf55438a5c212ad3R2107) is the usage of the set variable.
But in all other occurrences it's used as a point.

It's a little bit sad that the original design is not straight forward. :cry:
If we want to clarify this, we would need two different variables and we would adjust loading/saving to stay compatible with vanilla (check mode and if mode is PM_WALK use `Displacement` instead of `Point` variable).
But imho the gain is not worth the trouble.
If you want you can write a comment to explain this conversion. 🙂 

Note:
In Monster they even used another set of variables. 😕

Reply:
Fair enough. Will leave this as-is then as the only exception. I'm actually happy this refactor applied smoothly to all other parts of the code, though.